### PR TITLE
BUD-134 show view activity when joined

### PR DESCRIPTION
**Pull request by noahtallen**
This adds the ability to see the activity details while you are a member.

- Animates up when you join
- Animates down when you tap the mini description at the top
- Animates up when you tap the little up button
- See date, location, and picture in the mini description area.

The architecture should be set up such that the next functionality (e.g. delete user) should be easy to add.

_Note: all of the view activity code is in its own subdirectory now per @lukemeierj's request_

To test:
1. pull code and run
2. Go to topics and select an activity
    - If you are a member, the chat view should indicate that
    - If you are an owner, the chat view should indicate that
    - You should see a mini area if you have joined, and tap it to reveal the description (with a "close" button instead of a join button)
    -  If you are NOT a member of the activity, it should still show the maximized description and allow you to join it. On joining, there should be an animation into the joined activity view.

### CODE REVIEW

**Comment by noahtallen** [comment]
Note: I'd like to wait to merge until #55 is done - this was based off of that.

**Comment by lukemeierj** [comment]
Looks amazing!   Thanks for the folder biz

Report user is not supposed to work currently, right?  It logs "default" to the console.

File: Buddies/Activities/View Activity/ActivityDescriptionController.swift [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+            self.shrinkButton.isHidden = false
+        } else {
+            self.joinButton.isHidden = false
+            self.shrinkButton.isHidden = true
+        }
+        
+        // Conditionally handle hiding/showing
+        // the expanded description view:
+        if (shouldExpand) {
+            self.expandMe()
```
Comment (lukemeierj) [inline Buddies/Activities/View Activity/ActivityDescriptionController.swift:114]: ;)

**Review by lukemeierj — APPROVED** [review]
LGTM

Maybe we could talk about how I could reuse your Topic collection in the ProfileVC.  I think they're similar enough.

File: Buddies/Activities/View Activity/ActivityDescriptionController.swift (hunk 1 above)
Comment (noahtallen) [reply Buddies/Activities/View Activity/ActivityDescriptionController.swift:114]: 😉

**Comment by noahtallen** [comment]
@lukemeierj right on - none of that functionality (including removing members) is wired up to do anything yet. That's my next step :)

> Maybe we could talk about how I could reuse your Topic collection in the ProfileVC. I think they're similar enough.

Yes, I think that'd be fantastic! You could easily modify it to include a favoriting thing, and that could integrate easily into the view activity thing as well, I hope

**Pull request merged**